bento_lib/discovery/models/provenance/dataset.py

```diff
@@ -0,0 +1,202 @@
+    "Co-Investigator",
+    "Sub-Investigator",
+    "Study Director",
+    "Project Lead",
+    # Research team
+    "Researcher",
+    "Research Assistant",
+    "Data Scientist",
+    "Statistician",
+    "Study Coordinator",
+    "Lab Technician",
+    # Participants / human subjects
+    "Participant",
+    "Subject",
+    "Volunteer",
+    # Organizational / institutional roles
+    "Sponsoring Organization",
+    "Collaborating Organization",
+    "Institution",
+    "Site",
+    "Research Center",
+    "Publisher",  # DCAT
+    # Ethics & compliance
+    "IRB",
+    "Ethics Board",
+    "Data Monitoring Committee",
+    "Compliance Officer",
+    # Funding & support
+    "Sponsor",
+    "Funder",
+    "Grant Agency",
+    # Contributors (non-research)
+    "Consultant",
+    "Advisor",
+    "Reviewer",
+    # Data & technical roles
+    "Data Provider",
+    "Data Controller",
+    "Data Processor",
+    "Data Contributor",
+    # External stakeholders
+    "Partner",
+    "Stakeholder",
+    "Community Representative",
+    "Other",
+]
+
+# Derived from PCGL Study Model
+type StudyDomain = Literal[
+    "Aging",
+    "Birth Defects",
+    "Cancer",
+    "Circulatory and Respiratory Health",
+    "General Health",
+    "Infection and Immunity",
+    "Musculoskeletal Health and Arthritis",
+    "Neurodevelopmental Conditions",
+    "Neurosciences, Mental Health and Addiction",
+    "Nutrition, Metabolism and Diabetes",
+    "Population Genomics",
+    "Rare Diseases",
+    "Other",
+]
+
+type PublicationType = Literal[
+    "Journal Article",
+    "Preprint",
+    "Conference Paper",
+    "Book Chapter",
+    "Technical Report",
+    "Thesis",
+    "Dataset",
+]
+
+
+class Other(BaseModel):
+    """When a literal is not exhaustive"""
+
+    other: str
+
+
+class Phone(BaseModel):
+    country_code: int
+    number: int
+    extension: int | None
+
+
+class Contact(BaseModel):
+    email: list[str]
+    address: str | None
+    phone: Phone | None
+
+
+class Organization(BaseModel):
+    name: str
+    description: str | None
+    contact: Contact
+    role: list[Role]
+
+    # For funders
+    grant_number: str | None
+
+
+class Person(BaseModel):
+    first_name: str
+    last_name: str
+    honorific: str | None
+    other_names: list[str]
+
+    affiliations: list[Organization | str]
+
+    role: list[Role]
+
+
+class ParticipantCriteria(BaseModel):
+    type: Literal["Inclusion", "Exclusion"]
+    description: str
+
+
+class Count(BaseModel):
+    count_entity: str
+    value: int | float
+    description: str
+
+
+class License(BaseModel):
+    """Derived from DCAT"""
+
+    label: str
+    type: str
+    url: HttpUrl
+
+
+class Publication(BaseModel):
+    """
+    Publication or related resource link with metadata.
+    """
+
+    title: str
+    url: HttpUrl
+    doi: str | None
+    publication_type: PublicationType | Other
+    authors: list[str] | None
+    publication_date: date | None
+    journal: str | None
```

Comment:
we may need to make this model more flexible; book chapters/theses for example do not have journals.